src/browser/atom-window.coffee

```diff
@@ -73,6 +73,9 @@ class AtomWindow
     @browserWindow.loadUrl @getUrl(loadSettings)
     @browserWindow.focusOnWebView() if @isSpec
 
+    if locationsToOpen.length is 1 and locationsToOpen[0].pathToOpen is null
```

Comment:
Yeah, I'd like to. This is the beginning: https://github.com/atom/atom/blob/8fac0e87788aaf398adee0bb75f034ef497eb971/src/browser/atom-application.coffee#L93

I will look at propagating this with fresh eyes tomorrow. It seemed there would be some messiness in AtomApplication::openPath(). It's expecting a path. I could split out the [window creation things](https://github.com/atom/atom/blob/8fac0e87788aaf398adee0bb75f034ef497eb971/src/browser/atom-application.coffee#L378-L385) to another method. It just all felt dirty. This was the quickest thing I could get out to show the direction I wanted to take.